solr/contrib/velocity/src/java/org/apache/solr/response/VelocityResponseWriter.java

```diff
@@ -231,25 +222,32 @@ private VelocityContext createContext(SolrQueryRequest request, SolrQueryRespons
     resourceTool.configure(toolConfig);
     context.put("resource", resourceTool);
 
-/*
-    // Custom tools, specified in config as:
-        <queryResponseWriter name="velocityWithCustomTools" class="solr.VelocityResponseWriter">
-          <lst name="tools">
-            <str name="mytool">com.example.solr.velocity.MyTool</str>
-          </lst>
-        </queryResponseWriter>
-*/
-    // Custom tools can override any of the built-in tools provided above, by registering one with the same name
-    for(String name : customTools.keySet()) {
-      Object customTool = SolrCore.createInstance(customTools.get(name), Object.class, "VrW custom tool: " + name, request.getCore(), request.getCore().getResourceLoader());
-      if (customTool instanceof LocaleConfig) {
-        ((LocaleConfig)customTool).configure(toolConfig);
+   if (request.getCore().getCoreDescriptor().isConfigSetTrusted()) {
+      // Load custom tools, only if in a trusted configset
+
+      /*
+          // Custom tools, specified in config as:
+              <queryResponseWriter name="velocityWithCustomTools" class="solr.VelocityResponseWriter">
+                <lst name="tools">
+                  <str name="mytool">com.example.solr.velocity.MyTool</str>
+                </lst>
+              </queryResponseWriter>
+      */
+      // Custom tools can override any of the built-in tools provided above, by registering one with the same name
+      if (request.getCore().getCoreDescriptor().isConfigSetTrusted()) {
```

Comment:
Looks like it checks `request.getCore().getCoreDescriptor().isConfigSetTrusted()` twice. I suppose one check may be removed.

Reply:
Yes, that's right @artem-smotrakov . Actually I tried to align the code according to the original patch https://github.com/apache/lucene-solr/commit/9dfee35b9facbcbc9342d8bbbe52667cb6d1dced.  It was also the same there :) @erikhatcher , should I change it or keep it as it is?